### ISSUE TITLE
Get Current Selection Service no longer works

### ISSUE THREAD

**Issue by skurfer** [quicksilver/Quicksilver#2646]
**Describe the bug**

I noticed ⌘⎋ wasn’t working. I checked the Services prefs and saw “033” instead of the expected shortcut. I wiped it and attempted to set it again, but the system doesn’t allow you to use ⎋ in shortcuts. (I don’t remember if it used to.)

**Screenshots**

<img width="780" alt="Screen Shot 2022-02-17 at 9 00 42 AM" src="https://user-images.githubusercontent.com/154676/154497302-b245108a-3302-4dfd-b1ec-3e724ca54707.png">

**What version of macOS?**

12.2.1

**What version of Quicksilver?**

It’s complicated. 😂

**Comment by pjrobertson** [comment]
#complicated 😆 

Works fine on 10.14 for me. Appears to be a regression due to macOS 12 (potentially 11)

![Screenshot 2022-02-17 at 22 18 06](https://user-images.githubusercontent.com/150431/154500265-029b2280-2616-4bda-b157-7bcd7b371200.png)

**Comment by pjrobertson** [comment]
At the very least, the timeout shouldn't be 10s, that's way too long in this day and age. Let's make it 3s.

![Screenshot 2022-02-17 at 22 22 52](https://user-images.githubusercontent.com/150431/154501229-450d7697-82f6-4891-8cf5-88d6e130eef5.png)

**Comment by n8henrie** [comment]
Same -- `033` and won't let me use <kbd>Esc</kbd> after it's cleared. 12.2.1.

Oddly it still works fine in my stable build of QS -- even with the `Keyboard` -> `Shortcuts` option unticked / disabled.

It looks like it's coming from `Triggers` -> `Quicksilver` -> `Command Window with Selection`, which still was showing <kbd>cmd</kbd><kbd>Esc</kbd>.

Unfortunately after clearing it, when I try to re-insert <kbd>Esc</kbd>, it gives me a `5` instead. ?

**Comment by pjrobertson** [comment]
> This is me trying to re-input cmdEsc on a debug build:

Good catch, this is likely an ndhotkey bug. I'll look into it.

The only other change I can think of is the change in 6c5fae942186a66644ce218264078125b54b2b0e that jumbled up the plist, but the before and after look the same:

Before:
![Screenshot 2022-02-18 at 08 07 30](https://user-images.githubusercontent.com/150431/154592879-2ec0d85d-5c57-4e57-b882-bfea452c0754.png)

After:
![Screenshot 2022-02-18 at 08 07 55](https://user-images.githubusercontent.com/150431/154592892-ec12d278-845e-44be-92da-9b18ae150d06.png)

**Comment by n8henrie** [comment]
`\033` is 27 in octal: 

```
$ py '0o33'
27
```

27 is the MacOS keycode for escape:

```console
$ grep -i escape /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Carbon.framework/Versions/A/Frameworks/HIToolbox.framework/Versions/A/Headers/Events.h
  kEscapeCharCode               = 27,
  kVK_Escape                    = 0x35,
```

In triggers.plist, the key code for command is:

```
<integer>1048840</integer>
```

and `b` (which I used as an example) is: `<integer>11</integer>`

```python
>>> int("b", 16)
11
>>> hex(11)
'0xb'
>>> hex(1048840)
'0x100108'
```

My `Triggers.plist` has `<integer>53</integer>` listed for `<string>Command Window with Selection</string>`

```python
>>> hex(53)
'0x35'
```

```console
$ grep '35\b' /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Carbon.framework/Versions/A/Frameworks/HIToolbox.framework/Versions/A/Headers/Events.h
  kVK_Escape                    = 0x35,
```

So using <kbd>Esc</kbd> as an example, QS is displaying octal format (`\033`) and storing int format (`53`) in `Triggers.plist`; I imagine the `5` I'm seeing in my screenshot above is just a truncation of `53`.

**Comment by pjrobertson** [comment]
I didn't really need to do the `git bisect`... it just confirmed what we already knew:

```
64b8a916ae5b9a5d174e4ed6202c7d33111e6d41 is the first bad commit
commit 64b8a916ae5b9a5d174e4ed6202c7d33111e6d41
Author: Patrick Robertson <robertson.patrick@gmail.com>
Date:   Wed Feb 9 10:11:52 2022 +0800

    Ndhotkey fixes (#2594)
    
    * Add NDHotKeyEvent as a submodule.
    
    * Use latest NDHotKeyEvent
    Fixes #2405 Fixes #1134  - #1147 can be closed
```

I think the fix should be easy - just use ⌘⎋ in the plist.

**Comment by pjrobertson** [comment]
Yep, just required changing the plist key code to `⎋`. Spent ages debugging, even dove into Etienne's code from 8 years ago. These things always end up being so simple.

Will submit a PR
